### PR TITLE
Update with v12.0.0 libraries

### PR DESCRIPTION
**Pull request by torressa**
- [x] Update conda URLs
- [x] Update archive checksums
- [x] Update tree checksums
- [x] Update libs
- [x] Update tests
- [x] Remove `open_documentation` command (12.0 no longer ships the documentation)
- [x] Update README (OSS licenses are in a different location in the conda packages)

### CODE REVIEW

**Comment by simonbowly** [comment]
@torressa I rebased your PR and updated the artifacts for 12.0.0.

@odow I think this is ready to go? One notable change though is that the `open_documentation` command had to go away. The docs are not contained in the conda packages anymore.

File: test/runtests.jl [hunk 1]
```diff
@@ -42,7 +42,9 @@ end
 
 @testset "license_error" begin
     envptr = Ref{Ptr{Cvoid}}()
-    error = @ccall libgurobi.GRBemptyenv(envptr::Ptr{Ptr{Cvoid}})::Cint
+    # Temporary workaround for 12.0.0. This can be reverted to use GRBemptyenv for 12.0.1
+    # https://docs.gurobi.com/projects/optimizer/en/12.0/reference/releasenotes/knownbugs.html
+    error = @ccall libgurobi.GRBemptyenvinternal(envptr::Ptr{Ptr{Cvoid}}, 12::Int, 0::Int, 0::Int)::Cint
```
[the remark this answers is not in the document]
Comment (simonbowly) [reply test/runtests.jl:47]: `GRBemptyenv` is exposed via a header define. The majority of users don't need the internal function.

Can you clarify why it's a blocker here? libgrb_api.jl has the necessary additions, picked up automatically from the header files:

https://github.com/jump-dev/Gurobi.jl/blob/52ca3d9292804e0eea8f7f4e1a55eb517898c600/src/gen120/libgrb_api.jl#L4645-L4652

We've been testing Gurobi.jl with 12 already, and this doesn't appear to be a problem.

[the remark this answers is not in the document]
Comment (odow) [reply test/runtests.jl:47]: Ah, I didn't look for the header change. So this is a problem only for this particular test in Gurobi_jll.

[the remark this answers is not in the document]
Comment (simonbowly) [reply test/runtests.jl:58]: Correct

**Pull request merged**